CDCgov/prime-reportstream
# 8093 UP FHIR bundles should reference Diagnostic Reports in Provenance target property

This PR adjusts the HL7 to FHIR converter to add a reference to DiagnosticReports as a Provenance target in order to comply with the FHIR spec. The UP translate function is also modified to identify Endpoint resources as receivers only, maintaining its current behavior.

Test Steps:
1. Set a breakpoint in `HL7toFhirTranslator.kt` in the `translate` function.
2. Submit HL7 data to the UP.
3. Observe that a `target` exists in `Provenance` by the end of the function. You can call ``FhirTranscoder.encode(bundle)`` to output the content of the bundle in JSON.

## Changes
- Add functions in FHIRBundleHelpers to search for and insert DiagnosticReports in a bundle as Provenance targets.
- Add calls to the new helpers to ensure that all HL7 to FHIR conversions include a Provenance target.
- Adjust the UP translate function to only process endpoints as receivers.

## Checklist

### Testing

## Linked Issues
- Fixes #8093
